manticore/core/smtlib/expression.py

```diff
@@ -603,11 +607,12 @@ def index(self):
     def byte(self):
         return self.operands[2]
 
-class ArrayProxy(ArrayVariable):
+class ArrayProxy(Array):
```

Comment:
Why the change and not just keep `ArrayVariable` and the name it provides?

Reply:
With current implementation after you assign some key-value to this ArrayProxy it stops being an ArrayVariable. And it has no name assigned to it.


X = ArrayVariable()
Y = Store(X, an_address, a_value)  <- this is an Array without a name assigned to it. 

As an alternative implementation we can still return a *.name* for an ArrayProxy by creating a new variable when needed. Though it needs access to the constrain store from the ArrayProxy.
X = ArrayVariable()
Y = Store(X, an_address, a_value)
Z = ArrayVariable()
assert( Z == Y )